feature_engine/datetime/datetime.py

```diff
@@ -209,7 +211,14 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         # check if datetime variables contains na
         if self.missing_values == "raise":
-            _check_contains_na(X, self.variables_)
+            if self.variables_ == ["index"]:
```

Comment:
my idea was that the argument `variables` in the init, can take 3 values:

- None: defaults to finding all dt variables (already implemented)
- Variable list: checks if variables entered by user are or can be datetime (already implemented)
- "index", taken as a string, which will extract datetime features from the index (new in this PR)

Then, here I would tailor the message to say, there were nan in the index, for example.